packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -195,6 +226,61 @@ class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
     super.dispose();
   }
 
+  /// Detect spike in memory usage if so do an automatic snapshot.
+  void autoSnapshot({debugMode = false}) {
```

Comment:
rather than passing in debugMode, consider adding a top level field in this file called debugSnapshots or similar? that matches how flutter handles this sort of situation.

Reply:
Done.